modules/datahandling/filerequests.go

```diff
@@ -136,8 +380,43 @@ func (f *filePullRequest) setAbstractRequest(req *abstractRequest) {
 	f.abstractRequest = *req
 }
 
-func (f filePullRequest) process() (response *serverMessageWrapper, notification *serverMessageWrapper, err error) {
-	// TODO
-	fmt.Printf("Recieved file pull request from %s\n", f.SenderID)
-	return nil, nil, nil
+func (f filePullRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
+	// TODO (normal/required): check if permission high enough on project
+
+	res := new(serverMessageWrapper)
+	res.Timestamp = time.Now().UnixNano()
+	res.Type = "Responce"
+
+	res.ServerMessage = response{
+		Status: fail,
+		Tag:    f.Tag,
+		Data:   struct{}{}}
+
+	fileMeta, err := db.MySQLFileGetInfo(f.FileID)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	rawFile, err := db.FileRead(fileMeta.RelativePath, fileMeta.Filename, fileMeta.ProjectID)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	changes, err := db.CBGetFileChanges(f.FileID)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	res.ServerMessage = response{
+		Status: success,
+		Tag:    f.Tag,
+		Data: struct {
+			FileBytes []byte
+			Changes   []string
+		}{
+			FileBytes: *rawFile,
+			Changes:   changes,
```

Comment:
Could you add a todo/note about this? You can assign it to me if you want:

TODO (obsessiveorange): ...